recipes/libffi/all/conanfile.py

```diff
@@ -25,161 +28,157 @@ class LibffiConan(ConanFile):
+            env.define("NM", "dumpbin -symbols")
+            env.define("OBJDUMP", ":")
+            env.define("RANLIB", ":")
+            env.define("STRIP", ":")
+
+        tc.generate(env)
+
+        ms = VirtualBuildEnv(self)
+        ms.generate(scope="build")
+
+    def _patch_source(self):
+        apply_conandata_patches(self)
+
+        if Version(self.version) < "3.3":
+            if self.settings.compiler == "clang" and Version(str(self.settings.compiler.version)) >= 7.0:
+                # https://android.googlesource.com/platform/external/libffi/+/ca22c3cb49a8cca299828c5ffad6fcfa76fdfa77
+                sysv_s_src = os.path.join(self.source_folder, "src", "arm", "sysv.S")
+                replace_in_file(self, sysv_s_src, "fldmiad", "vldmia")
+                replace_in_file(self, sysv_s_src, "fstmiad", "vstmia")
+                replace_in_file(self, sysv_s_src, "fstmfdd\tsp!,", "vpush")
+
+                # https://android.googlesource.com/platform/external/libffi/+/7748bd0e4a8f7d7c67b2867a3afdd92420e95a9f
+                replace_in_file(self, sysv_s_src, "stmeqia", "stmiaeq")
 
     def build(self):
-        self._patch_sources()
-        shutil.copy(self._user_info_build["gnu-config"].CONFIG_SUB,
-                    os.path.join(self._source_subfolder, "config.sub"))
-        shutil.copy(self._user_info_build["gnu-config"].CONFIG_GUESS,
-                    os.path.join(self._source_subfolder, "config.guess"))
-
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.make()
-            if tools.get_env("CONAN_RUN_TESTS", False):
-                autotools.make(target="check")
+        self._patch_source()
+        autotools = Autotools(self)
+        autotools.configure()
+        autotools.make()
 
     def package(self):
-        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
-        if self._is_msvc:
-            if self.options.shared:
-                self.copy("libffi.dll", src=".libs", dst="bin")
-            self.copy("libffi.lib", src=".libs", dst="lib")
-            self.copy("*.h", src="include", dst="include")
-        else:
-            with self._build_context():
-                autotools = self._configure_autotools()
-                autotools.install()
-
-            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-            tools.rmdir(os.path.join(self.package_folder, "share"))
-            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
+        autotools = Autotools(self)
+        autotools.install(args=[f"DESTDIR={unix_path(self, self.package_folder)}"])  # Need to specify the `DESTDIR` as a Unix path, aware of the subsystem
+        fix_apple_shared_install_name(self)
+        copy(self, "*.dll", os.path.join(self.package_folder, "lib"), os.path.join(self.package_folder, "bin"))
```

Comment:
but it duplicates the dll. They should be moved, not copied.

Reply:
Yeah, that's bad. I've pushed a fix now.